src/Components/SingleProductPage/ProductImageSection/ProductImageSection.jsx

```diff
@@ -0,0 +1,45 @@
+import { useWishlist } from "../../../Context";
+
+const ProductImageSection = ({ item }) => {
+  const { wishlist, setWishlist } = useWishlist();
+  const { src } = item;
+
+  const addWishlistClick = () => {
+    setWishlist((oldCart) => {
+      return [...new Set([...oldCart, item])];
+    });
+  };
+
+  const removeFromWishlist = () => {
+    setWishlist((oldWishlist) => {
+      return oldWishlist.filter((el) => {
+        return el._id !== item._id;
+      });
+    });
+  };
+
+  return (
+    <div className="single-product-image">
+      <button
+        onClick={
+          wishlist.includes(item) ? removeFromWishlist : addWishlistClick
```

Comment:
Instead of putting condition here, you can make a handler and put the condition in it. It would look much cleaner.